backend/endpoints/rooms.py

```diff
@@ -62,3 +63,95 @@ async def create_room(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="unable to create room",
     )
+
+
+@router.put(
+    "/org/{org_id}/rooms/{room_id}/members/{member_id}",
+    status_code=status.HTTP_200_OK,
+    responses={
+        400: {"detail": "the max number for a Group_DM is 9"},
+        401: {"detail": "member not an admin"},
+        403: {"detail": "DM room or not found"},
+        424: {"detail": "failed to add new members to room"},
+    },
+)
+async def add_to_room(
+    org_id: str,
+    room_id: str,
+    member_id: str,
+    data: AddToRoom,
+    background_tasks: BackgroundTasks,
+):
+    """Adds a new member(s) to a room
+    Args:
+        data: a pydantic schema that defines the request params
+        org_id (str): A unique identifier of an organisation
+        room_id: A unique identifier of the room to be updated
+        member_id: A unique identifier of the member initiating the request
+        background_tasks: A parameter that allows tasks to be performed outside of the main function
+
+    Returns:
+        HTTP_200_OK: {
+                        "status": 200,
+                        "message": "success",
+                        "data": {
+                            "matched_documents": 1,
+                            "modified_documents": 1
+                        }
+                    }
+    Raises:
+        HTTP_400_BAD_REQUEST: the max number for a Group_DM is 9
+        HTTP_401_UNAUTHORIZED: member not in room or not an admin
+        HTTP_403_FORBIDDEN: DM room or not found
+        HTTP_424_FAILED_DEPENDENCY: failed to add new members to room
+    """
+
+    DB = DataStorage(org_id)  # initializes the datastorage class with the org id
+    new_member = data.dict().get("new_member")
+    room = await get_room(org_id=org_id, room_id=room_id)
+
+    if not room or room["room_type"] == RoomType.DM:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="DM room or not found",
+        )
+
+    member = room.get("room_members").get(str(member_id))
+    if member is None or member["role"] != Role.ADMIN:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="member not in room or not an admin",
+        )
+
+    if room["room_type"] == RoomType.CHANNEL:
+        room["room_members"].update(new_member)
+
+    if room["room_type"] == RoomType.GROUP_DM:
+        room["room_members"].update(new_member)
+        if len(room["room_members"].keys()) > 9:
+            raise HTTPException(
```

Comment:
Would it not be better to do the test and throw this exception before adding the member to the room? Doesn't `update` persist the new member already?

i.e. `if len(room["room_members"].keys()) >= 9:` then throw the exception first.

Reply:
The proces @mykie88 is following is that he's updating the members_list.
Then doing the test to confirm that even with the updated members the count doesn't still exceed 9

Then if the test passes, he persist the data.

Reply:
no @Aniebet that update is just a dict method. The data is persisted further down the code at the DB.update method of the datastorage class

Reply:
@almaudoh